WalletWasabi.Fluent/ViewModels/Wallets/Send/SendViewModel.cs

```diff
@@ -307,7 +307,7 @@ protected override void OnNavigatedTo(bool inHistory, CompositeDisposable dispos
 
 			if (_coinJoinManager is { } coinJoinManager)
 			{
-				coinJoinManager.IsUserInSendWorkflow = true;
+				coinJoinManager.WalletEnteredSendWorkflow(_wallet);
```

Comment:
Why are you passing the wallet here? The only thing that CoinjoinManager is tracking is the name. This breaks The Principle of Least Knowledge (The Law of Demeter) that says that a function should receive what it needs and no more than that. For example, if you have to code a function that sums seconds an return minutes, this is the correct signature:

`float Sum(int seconds1, int seconds2)
` 

And this is the incorrect signature:

`float Sum(DateTimeOffset date1, DateTimeOffset data2)
`

Why? Because it creates unnecessary dependencies and makes the code less reusable while incentivizes the abuse, basically someone can say: Hey! look, I have access to the wallet here so, lets use that.

This is everywhere in Wasabi and we have to stop one day so, lets start today.